src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.java

```diff
@@ -604,12 +604,31 @@ private KfsVendorDataWrapper buildRemainingVendorNotes(PaymentWorksVendor pmwVen
     }
     
     private KfsVendorDataWrapper createGoodsAndServicesNote(PaymentWorksVendor pmwVendor, KfsVendorDataWrapper kfsVendorDataWrapper) {
-        StringBuilder sbText = new StringBuilder(getConfigurationService().getPropertyValueAsString(PaymentWorksKeyConstants.NEW_VENDOR_PVEN_NOTES_GOODS_AND_SERVICES_PROVIDED_LABEL)).append(KFSConstants.BLANK_SPACE).append(pmwVendor.getServicesProvided());
+        StringBuilder sbText = new StringBuilder(getConfigurationService().getPropertyValueAsString(PaymentWorksKeyConstants.NEW_VENDOR_PVEN_NOTES_GOODS_AND_SERVICES_PROVIDED_LABEL)).append(KFSConstants.BLANK_SPACE).append(pmwVendor.getServicesProvided()).append(System.lineSeparator());
+        if (servicesAreBeingProvided(pmwVendor)) {
+            sbText = populateAnswersToServiceQuestions(sbText, pmwVendor);
+        }
         Note goodsServicesNote = getPaymentWorksBatchUtilityService().createNote(sbText.toString());
         kfsVendorDataWrapper.getVendorNotes().add(goodsServicesNote);
         return kfsVendorDataWrapper;
     }
     
+    private boolean servicesAreBeingProvided(PaymentWorksVendor pmwVendor) {
+        return (StringUtils.isNotBlank(pmwVendor.getServicesProvided())
+                && (StringUtils.equalsIgnoreCase(pmwVendor.getServicesProvided(), PaymentWorksConstants.PaymentWorksGoodsVsServicesOptions.SERVICES.getOptionValueAsString())
+                    | StringUtils.equalsIgnoreCase(pmwVendor.getServicesProvided(), PaymentWorksConstants.PaymentWorksGoodsVsServicesOptions.GOODS_WITH_SERVICES.getOptionValueAsString())));
+    }
+    
+    private StringBuilder populateAnswersToServiceQuestions(StringBuilder sbText, PaymentWorksVendor pmwVendor) {
+        sbText.append(getConfigurationService().getPropertyValueAsString(PaymentWorksKeyConstants.NEW_VENDOR_PVEN_NOTES_GOODS_AND_SERVICES_CURRENT_PAYROLL_PAID_LABEL)).append(KFSConstants.BLANK_SPACE).append((pmwVendor.isCurrentlyPaidThroughPayroll() ? KFSConstants.OptionLabels.YES : KFSConstants.OptionLabels.NO)).append(System.lineSeparator()).append(System.lineSeparator());
```

Comment:
These six answer-populating lines have a lot of similarities between them. To reduce duplication, it would be nice to have a separate method to do the actual text-appending, and then supply arguments for the differences accordingly (such as the key for the config property, and a lambda expression or method reference to evaluate the vendor boolean property).